### ISSUE TITLE
Error and server shutdown on http request.

### ISSUE THREAD

**Issue by Miro001** [triton-inference-server/server#4061]
**Description**
Server is launched without errors and is serving model with custom ops from tensorflow-text. (looks like it)
We are importing custom op from tensorflow-text, specifically the one that is causing errors is: 
> /usr/local/lib/python3.8/dist-packages/tensorflow_text/python/ops/_regex_split_ops.so

Http fails to respond 

`curl -v localhost:8000/v2/health/ready`

```
*   Trying ::1...
* TCP_NODELAY set
* Expire in 150000 ms for 3 (transfer 0x563b1ccf3fb0)
* Expire in 200 ms for 4 (transfer 0x563b1ccf3fb0)
* Connected to localhost (::1) port 8000 (£0)
> GET /v2/health/ready HTTP/1.1
> Host: localhost:8000
> User-Agent: curl/7.64.0
> Accept: */*
* Empty reply from server
* Connection £0 to host localhost left intact
curl: (52) Empty reply from server
```
and server shutdown with an error:

```
I0316 13:52:47.321972 1 http_server.cc:2976] HTTP request: 0 /v2/health/ready
Signal (11) received.
 0# 0x000055F5850AD299 in tritonserver
 1# 0x00007F9E6FB3F210 in /usr/lib/x86_64-linux-gnu/libc.so.6
 2# re2::RE2::Match(re2::StringPiece const&, unsigned long, unsigned long, re2::RE2::Anchor, re2::StringPiece*, int) const in /usr/lib/x86_64-linux-gnu/libre2.so.5
 3# re2::RE2::DoMatch(re2::StringPiece const&, re2::RE2::Anchor, unsigned long*, re2::RE2::Arg const* const*, int) const in /usr/lib/x86_64-linux-gnu/libre2.so.5
 4# 0x000055F5850C0658 in tritonserver
 5# 0x000055F5852A2CF6 in tritonserver
 6# 0x000055F5852A7B78 in tritonserver
 7# 0x000055F5852A595E in tritonserver
 8# 0x000055F58528C8F0 in tritonserver
 9# 0x000055F585294410 in tritonserver
10# 0x000055F585294E5F in tritonserver
11# 0x000055F5852A97B2 in tritonserver
12# 0x00007F9E703AB609 in /usr/lib/x86_64-linux-gnu/libpthread.so.0
13# clone in /usr/lib/x86_64-linux-gnu/libc.so.6
```
This does not happen without importing this specific custom op:
> /usr/local/lib/python3.8/dist-packages/tensorflow_text/python/ops/_regex_split_ops.so

**Note:**
We can actually make requests through GRPC for metadata and inference. HTTP does not work. Probably due to the regex library issue.

**Triton Information**

nvcr.io/nvidia/tritonserver:22.02-tf2-python-py3, but reproduced with multiple version 21.09->21.12

**To Reproduce**
1. Create a dockerfile
```
FROM nvcr.io/nvidia/tritonserver:22.02-tf2-python-py3

RUN  pip install tensorflow-text==2.7.3 --no-dependencies
```
2. Build a custom version of triton
`docker build . -f Dockerfile.Triton --tag custom-triton`

3. Generate any model (jsut for debug) and save it in recommended file structure.
```
import numpy as np
import tensorflow as tf
from tensorflow import keras

def create_model():
  model = tf.keras.models.Sequential([
    keras.layers.Dense(512, activation='relu', input_shape=(784,)),
    keras.layers.Dropout(0.2),
    keras.layers.Dense(10)
  ])

  model.compile(optimizer='adam',
                loss=tf.losses.SparseCategoricalCrossentropy(from_logits=True),
                metrics=[tf.metrics.SparseCategoricalAccuracy()])

  return model

model = create_model()

model.save('./my_mount/my_model/1/model.savedmodel')
```
4. Check mount paths and launch triton server. Server is running 
```
docker run \
       --gpus=1 \
       --rm -p8000:8000 -p8001:8001 \
            -p8002:8002 \
            -v ~/my_mount:/models  -e "LD_LIBRARY_PATH=/opt/tritonserver/backends/tensorflow2:$LD_LIBRARY_PATH" \
            -e "LD_PRELOAD=/usr/local/lib/python3.8/dist-packages/tensorflow_text/python/ops/_normalize_ops.so:/usr/local/lib/python3.8/dist-packages/tensorflow_text/python/ops/_regex_split_ops.so:/usr/local/lib/python3.8/dist-packages/tensorflow_text/python/ops/_wordpiece_tokenizer.so:$LD_PRELOAD"\
       custom-triton:latest  tritonserver --model-repository=/models --strict-model-config=false --backend-config=tensorflow,version=2 
```
5.  Run curl request
`curl -v localhost:8000/v2/health/ready`
6. Get an error response
Error
**Expected behavior**
HTTP requests work.

**Comment by Miro001** [comment]
Thanks for the response.

- Does this library depend on RE2 as well? 
 Yes. It depends on RE2 as well.

- If so, can you check if it is built with the same version of RE2 in the docker container?
I was not able to find the exact build environment of this official pip package. I dont think it is available. However i tried to compile it myself  with official nvidia image (`nvcr.io/nvidia/tensorflow:22.02-tf2-py3`) and end up ABI errors (`*.so: undefined symbol: `) to control RE2 version.

**Comment by deadeyegoodwin** [comment]
You will need to ask Tensorflow: https://github.com/tensorflow/tensorflow

**Comment by erlichsefisalesforce** [comment]
@Miro001 facing the same issue. did you manage to solve this issue?

**Comment by Miro001** [comment]
@erlichsefisalesforce Long story short; Libs that you are trying to link are either incompatible with triton server or with your tf backend. With the way they are imported, Libs are shared (sucks) . You have several  options to resolve this:

1. Split your model to work with default tf backend ( recommended) - Get rid of whatever tensorflow_text is doing in your model and do it in separate service. I assume its preprocessing so it should be easy. You can even try triton with python runtime.
2. Compile your tf backend with tensorflow_text (tricky). You wont link it through LD_PRELOAD so you can use separate RE2 versions for tf backend and triton server.

**Comment by erlichsefisalesforce** [comment]
Thanks, @Miro001! after some more time I found that the error is propagated from the metric port of the triton server. blocking it prevented the server to crash. in my case, I was able to disable to metric port.

**Comment by sergeywong** [comment]
@erlichsefisalesforce I faced the same issue, too. I built the triton-server from source code. In may case, I pre-installed re2 (https://github.com/google/re2.git) which conflicted with the version in third_party of grpc (https://github.com/grpc/grpc/tree/v1.48.0/third_party). It's solved using the aforementioned version in grpc.  Maybe it's irrelevant to metric port in your case.
